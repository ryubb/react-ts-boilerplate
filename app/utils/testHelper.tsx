import * as React from "react";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import { render as testingLibRender } from "@testing-library/react";
import "jest-styled-components";
import configureStore from "redux-mock-store";
import { initialState } from "../redux/rootReducer";

const mockStore = configureStore();

type TargetType = React.ComponentElement<any, any> | React.ElementType;

type Props = {
  target: TargetType;
  initialState: Record<string, unknown>;
  path?: Array<string>;
};

export const TargetWithMockState: React.ElementType = ({
  target,
  initialState,
  path,
}: Props) => {
  const MockedTarget: React.ElementType = () => (
    <Provider store={mockStore(initialState)}>{target}</Provider>
  );

  return path ? (
    <MemoryRouter initialEntries={path}>
      <MockedTarget />
    </MemoryRouter>
  ) : (
    <MockedTarget />
  );
};

export const createBasicSnapshotTest = (
  target: TargetType,
  path?: Array<string>,
  state?: Record<string, unknown> | Array<unknown>
) => {
  it("should match snapshot via react-test-renderer.", () => {
    const tree = renderer
      .create(
        <TargetWithMockState
          target={target}
          path={path}
          initialState={{ ...initialState, ...state }}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
};

export const render = (
  target: TargetType,
  path?: Array<string>,
  state?: Record<string, unknown> | Array<unknown>
) =>
  testingLibRender(
    <TargetWithMockState
      target={target}
      path={path}
      initialState={{ ...initialState, ...state }}
    />
  );
