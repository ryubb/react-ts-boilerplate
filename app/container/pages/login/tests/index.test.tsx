import * as React from "react";
import { createBasicSnapshotTest } from "../../../../utils/testHelper";
import LoginPage from "..";

const path = ["/login"];

describe("LoginPage", () => {
  createBasicSnapshotTest(<LoginPage />, path);
});
