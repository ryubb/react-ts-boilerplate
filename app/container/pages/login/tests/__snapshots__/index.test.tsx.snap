// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage should match snapshot via react-test-renderer. 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  max-width: 500px;
  width: 100%;
  height: 440px;
  background-color: #fff;
  margin: 100px 20px 0;
}

.c2 {
  max-width: 320px;
  width: 100%;
  margin: 0 auto;
  padding-top: 80px;
}

.c3 {
  margin-bottom: 40px;
  font-size: 20px;
  font-weight: bold;
}

.c4 {
  position: relative;
  margin-bottom: 40px;
}

.c5 {
  position: absolute;
  bottom: -20px;
  width: 100%;
  font-size: 12px;
  color: #000;
  text-align: right;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <h2
        className="c3"
      >
        ログイン
      </h2>
      <form
        onSubmit={[Function]}
      >
        <div
          className="c4"
        >
          <input
            name="email"
            type="email"
          />
        </div>
        <div
          className="c4"
        >
          <input
            name="password"
            type="password"
          />
          <a
            className="c5"
            href="/"
            onClick={[Function]}
          >
            パスワードを忘れた場合
          </a>
        </div>
        <button
          type="submit"
        >
          送信
        </button>
      </form>
    </div>
  </div>
</div>
`;
